rollingshutter.py

```diff
@@ -1,31 +1,46 @@
-def rs(folder,speed=1):
-    
-    from PIL import Image
-    import glob
-    
-    # ENTER YOUR DIRECTORIES AND FILE TYPE HERE
-    frame_dir = "/Users/mattparker/Documents/rollingshuttervideos/" + str(folder) + "/"
-    frame_file = "png"
-    output_dir = "/Users/mattparker/Documents/rollingshuttervideos/"
-    
-    width = 1920
-    height = 1080
+from PIL import Image
+import argparse
+import glob
+import os
+
+
+def rolling_shutter(input_directory, output_directory, extension_type, image_height, image_width, speed):
     
     # Making our blank output frame
-    output_image = Image.new('RGB', (width, height)) 
+    output_image = Image.new('RGB', (image_width, image_height)) 
     
     
     # let us go through the frames one at a time
     
     current_row = 0
-    
-    for filename in glob.glob(frame_dir + "*." + frame_file):
+     
+    for filename in glob.glob(os.path.join(input_directory, "*.", extension_type)):
         frame = Image.open(filename)
-        new_line = frame.crop((0, current_row, width, current_row+speed))
+        new_line = frame.crop((0, current_row, image_width, current_row + speed))
         output_image.paste(new_line, (0,current_row))
         current_row += speed
     
     # and export the final frame
-    output_image.save(output_dir + 'output_image.png')
+    if not os.path.exists(output_directory):
+            os.makedirs(output_directory)
+    output_image.save(os.path.join(output_directory, 'output_image.png'))
     
     return 'DONE'
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input_directory', required=True, help='Path to directory containing input images.')
```

Comment:
I dunno, "explicit is better than implicit" is usually my approach (I don't want to remember the order of arguments). Either way, I don't think it matters too much.